packages/field/test/DisabledMixin.test.js

```diff
@@ -0,0 +1,173 @@
+import { expect, fixture, html, defineCE, unsafeStatic } from '@open-wc/testing';
+
+import { LitElement } from '@lion/core';
+import { DisabledMixin } from '../src/DisabledMixin.js';
+
+describe('DisabledMixin', () => {
+  before(() => {
+    class DisabledAble extends DisabledMixin(LitElement) {}
+    customElements.define('disable-able', DisabledAble);
+  });
+
+  it('reflects disabled to attribute', async () => {
+    const el = await fixture(html`
+      <disable-able></disable-able>
+    `);
+    expect(el.hasAttribute('disabled')).to.be.false;
+    el.disabled = true;
+    await el.updateComplete;
+    expect(el.hasAttribute('disabled')).to.be.true;
+  });
+
+  it('can be requested to be disabled', async () => {
+    const el = await fixture(html`
+      <disable-able></disable-able>
+    `);
+    el.makeRequestToBeDisable();
+    expect(el.disabled).to.be.true;
+  });
+
+  it('disabled does not override user provided tabindex', async () => {
+    const el = await fixture(html`
+      <disable-able tabindex="5" disabled></disable-able>
+    `);
+    expect(el.getAttribute('tabindex')).to.equal('-1');
+    el.disabled = false;
+    await el.updateComplete;
+    expect(el.getAttribute('tabindex')).to.equal('5');
+  });
+
+  it('will not allow to become enabled after makeRequestToBeDisable()', async () => {
+    const el = await fixture(html`
+      <disable-able></disable-able>
+    `);
+    el.makeRequestToBeDisable();
+    expect(el.disabled).to.be.true;
+
+    el.disabled = false;
+    expect(el.disabled).to.be.true;
+  });
+
+  it('will stay disabled after retractRequestToBeDisable() if it was disabled before', async () => {
+    const el = await fixture(html`
+      <disable-able disabled></disable-able>
+    `);
+    el.makeRequestToBeDisable();
+    el.retractRequestToBeDisable();
+    expect(el.disabled).to.be.true;
+  });
+
+  it('will become enabled after retractRequestToBeDisable() if it was enabled before', async () => {
+    const el = await fixture(html`
+      <disable-able></disable-able>
+    `);
+    el.makeRequestToBeDisable();
+    expect(el.disabled).to.be.true;
+    el.retractRequestToBeDisable();
+    expect(el.disabled).to.be.false;
+  });
+
+  it('will restore last state after retractRequestToBeDisable()', async () => {
+    const el = await fixture(html`
+      <disable-able></disable-able>
+    `);
+    el.makeRequestToBeDisable();
+    el.disabled = true;
+    el.retractRequestToBeDisable();
+    expect(el.disabled).to.be.true;
+
+    el.makeRequestToBeDisable();
+    el.disabled = false;
+    el.retractRequestToBeDisable();
+    expect(el.disabled).to.be.false;
+  });
+
+  describe('Can manage TabIndex', () => {
+    it('has an initial tabIndex of 0', async () => {
```

Comment:
Why does it set tabindex=0 by default?
Is everything that gets disabled also a focussable element?

Reply:
The DisabledMixin only manages disabled by default. You can opt into also managing tabIndex which you probably only should do for elements you actually want to be focusable